tests/tests/tier0/bluechi-monitor-service-removed/test_bluechi_monitor_service_removed.py

```diff
@@ -0,0 +1,56 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import os
+import threading
+import time
+
+from typing import Dict
+
+from bluechi_test.test import BluechiTest
+from bluechi_test.machine import BluechiControllerMachine, BluechiAgentMachine
+from bluechi_test.config import BluechiControllerConfig, BluechiAgentConfig
+
+node_name_foo = "node-foo"
+service_simple = "simple.service"
+
+
+def monitor_service(ctrl: BluechiControllerMachine):
+    result, output = ctrl.run_python(os.path.join("python", "monitor.py"))
+    if result != 0:
+        raise Exception(output)
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+    node_foo = nodes[node_name_foo]
+    node_foo.copy_systemd_service(
+        service_simple, "systemd", os.path.join("/", "etc", "systemd", "system"))
+    assert node_foo.wait_for_unit_state_to_be(service_simple, "inactive")
+
+    ctrl.bluechictl.start_unit(node_name_foo, service_simple)
+    nodes[node_name_foo].wait_for_unit_state_to_be(service_simple, "active")
+
+    t = threading.Thread(target=monitor_service, args=(ctrl,))
+    t.start()
+
+    # Wait till monitoring thread on controller starts properly
+    time.sleep(0.5)
+
+    node_foo.systemctl.stop_unit(service_simple)
+    node_foo.exec_run(f"rm -f /etc/systemd/system/{service_simple}")
+    node_foo.systemctl.daemon_reload()
```

Comment:
The `UnitRemoved` signal is emitted when a unit is unloaded (see Signals section [here](https://www.freedesktop.org/wiki/Software/systemd/dbus/)). So removing the file and daemon-reload aren't necessary.

Reply:
Hmm, so I totally missed that and my interpretation of the event is wrong. So how can I invoke unit removed signal? Because according to my test it was invoked when I removed the file on the disk:

Service cups is running when monitoring is started:

```
[root@mperina-cs9 ~]# bluechictl monitor mperina-cs9-2 cups.service
Monitor path: /org/eclipse/bluechi/monitor/1
Subscribing to node 'mperina-cs9-2' and unit 'cups.service'
[mperina-cs9-2] cups.service
	Unit created (reason: virtual)
[mperina-cs9-2] cups.service
	Unit state changed (reason: virtual)
	Active: active (running)
[mperina-cs9-2] cups.service
	Unit properties changed (Interface: org.freedesktop.systemd1.Service)
	StatusText: Scheduler is running...
[mperina-cs9-2] cups.service
	Unit properties changed (Interface: org.freedesktop.systemd1.Unit)
	ActiveState: active
```

On package uninstallation the unit is stopped first:

```
[mperina-cs9-2] cups.service
	Unit properties changed (Interface: org.freedesktop.systemd1.Service)
	StatusText: Scheduler is running...
[mperina-cs9-2] cups.service
	Unit properties changed (Interface: org.freedesktop.systemd1.Unit)
	ActiveState: active
[mperina-cs9-2] cups.service
	Unit properties changed (Interface: org.freedesktop.systemd1.Service)
	StatusText: Scheduler is running...
[mperina-cs9-2] cups.service
	Unit state changed (reason: real)
	Active: deactivating (stop-sigterm)
[mperina-cs9-2] cups.service
	Unit properties changed (Interface: org.freedesktop.systemd1.Unit)
	ActiveState: deactivating
[mperina-cs9-2] cups.service
	Unit properties changed (Interface: org.freedesktop.systemd1.Service)
	StatusText: Scheduler is running...
[mperina-cs9-2] cups.service
	Unit state changed (reason: real)
	Active: inactive (dead)
[mperina-cs9-2] cups.service
	Unit properties changed (Interface: org.freedesktop.systemd1.Unit)
	ActiveState: inactive
```

And the service file is removed from disk:

```
[mperina-cs9-2] cups.service
	Unit properties changed (Interface: org.freedesktop.systemd1.Service)
	StatusText: Scheduler is running...
[mperina-cs9-2] cups.service
	Unit properties changed (Interface: org.freedesktop.systemd1.Unit)
	ActiveState: inactive
[mperina-cs9-2] cups.service
	Unit removed (reason: real)
[mperina-cs9-2] cups.service
	Unit created (reason: real)
[mperina-cs9-2] cups.service
	Unit removed (reason: real)
[mperina-cs9-2] cups.service
	Unit created (reason: real)
[mperina-cs9-2] cups.service
	Unit removed (reason: real)
```

So `Unit removed (reson: real)` was received only when the service file was removed from disk.

Do you know any other option how to raise UnitRemoved event?

Reply:
UnitCreated and UnitRemoved signals are also received when `systemctl status` is invoked on `inactive` service, adapting the test accordingly